### PR TITLE
Add integration tests and cleanup

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-774

### Change description ###

- Added integration tests for end to end flow. Used WireMock for CCD callback.

- Commeted out validator and ignored unit tests as it is currently validating using exception record case data rather than transformed case data.

- Removed case data helper integration test.

**Please note that there are no validation integration tests as there is bug which needs to be fixed first.**

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.bulkscancore.domain.CaseDetails;
+import uk.gov.hmcts.reform.sscs.bulkscancore.domain.ExceptionCaseData;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class CcdCallbackControllerTest {
+
+    private static final String AUTHORIZATION_HEADER_KEY = "Authorization";
```
Comment (timja) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java]: should be a spring or servlet constant you can use for this?

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+
+    private static final String SERVICE_AUTHORIZATION_HEADER_KEY = "ServiceAuthorization";
+
+    private static final String USER_AUTH_TOKEN = "Bearer TEST_USER_AUTH_TOKEN";
```
Comment (timja) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java]: you have bearer space in here but not in this service auth token one...
would suggest removing bearer from this and making a separate const and joining the two

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+    private TestRestTemplate restTemplate;
+
+    @LocalServerPort
+    private int randomServerPort;
+
+    @MockBean
+    private AuthTokenValidator authTokenValidator;
+
+    @ClassRule
+    public static WireMockClassRule ccdServer = new WireMockClassRule(WireMockSpring.options().port(4000)
```
Comment (timja) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java]: don't hardcode ports that means that anyone running on port 4000 for something else will have this test fail for them, you should be able to get it to select an available port or find one your self

REQ

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+        .bindAddress("localhost").notifier(new ConsoleNotifier(true)));
```
Comment (timja) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java]: doesn't it bind to localhost by default?

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+
+    private String baseUrl;
+
+    @Before
+    public void setUp() {
+        baseUrl = "http://localhost:" + randomServerPort + "/exception-record/";
```
Comment (timja) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java]: a little confused on why there's this port and the wiremock one

File: src/main/java/uk/gov/hmcts/reform/sscs/exceptionhandlers/ResponseExceptionHandler.java [hunk 6]
```diff
@@ -16,19 +17,26 @@
...
         log.error(exception.getMessage(), exception);
 
         return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
     }
 
+    @ExceptionHandler(FeignException.class)
+    protected ResponseEntity<Void> handleFeignException(FeignException exception) {
+        log.error(exception.getMessage(), exception);
+
+        return ResponseEntity.status(exception.status()).build();
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/sscs/exceptionhandlers/ResponseExceptionHandler.java]: any reason you aren't adding the exception message here?

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java (hunk 1 above)
Comment (nitinprabhu) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java]: Done fixed as part of commit

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java (hunk 2 above)
Comment (nitinprabhu) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java]: Done

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java (hunk 4 above)
Comment (nitinprabhu) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/controllers/CcdCallbackControllerTest.java]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/exceptionhandlers/ResponseExceptionHandler.java (hunk 6 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/sscs/exceptionhandlers/ResponseExceptionHandler.java]: added now

**Pull request merged**